agent/consul/state/acl_events.go

```diff
@@ -0,0 +1,54 @@
+package state
+
+import (
+	"github.com/hashicorp/consul/agent/consul/stream"
+	"github.com/hashicorp/consul/agent/structs"
+	memdb "github.com/hashicorp/go-memdb"
+)
+
+// ACLEventsFromChanges returns all the ACL token, policy or role events that
+// should be emitted given a set of changes to the state store.
+// TODO: Add OpDelete/OpUpdate to the event or payload?
```

Comment:
I've omitted the Op for now, as it is not used by the ACLs.

Since every Event will have some type of operation, I'm wondering if it would make sense to make these part of the general `Event` struct, instead of part of the payload. 

Would `Create`, `Update`, `Delete` handle all cases?